redbot/core/settings_caches.py

```diff
@@ -122,157 +122,145 @@ async def set_ignored_guild(self, guild: discord.Guild, set_to: bool):
 class WhitelistBlacklistManager:
     def __init__(self, config: Config):
         self._config: Config = config
-        self._cached_whitelist: Dict[Optional[int], List[int]] = {}
-        self._cached_blacklist: Dict[Optional[int], List[int]] = {}
+        self._cached_whitelist: Dict[Optional[int], Set[int]] = {}
+        self._cached_blacklist: Dict[Optional[int], Set[int]] = {}
 
-    async def get_whitelist(self, guild: Optional[discord.Guild] = None) -> List[int]:
-        ret: List[int]
+    async def get_whitelist(self, guild: Optional[discord.Guild] = None) -> Set[int]:
+        ret: Set[int]
 
         gid: Optional[int] = guild.id if guild else None
 
         if gid in self._cached_whitelist:
             ret = self._cached_whitelist[gid].copy()
         else:
             if gid is not None:
-                ret = await self._config.guild_from_id(gid).whitelist()
-                if not ret:
-                    ret = []
+                ret = set(await self._config.guild_from_id(gid).whitelist())
             else:
-                ret = await self._config.whitelist()
+                ret = set(await self._config.whitelist())
 
             self._cached_whitelist[gid] = ret.copy()
 
         return ret
 
-    async def add_to_whitelist(self, guild: Optional[discord.Guild], role_or_user: List[int]):
+    async def add_to_whitelist(self, guild: Optional[discord.Guild], role_or_user: Iterator[int]):
```

Comment:
Wrong type hint (also, look at the other comment below)
```suggestion
    async def add_to_whitelist(self, guild: Optional[discord.Guild], role_or_user: Iterable[int]):
```